platform/darwin/src/MGLConversion.h

```diff
@@ -85,7 +85,7 @@ class ConversionTraits<Holder> {
     static optional<float> toNumber(const Holder& holder) {
         const id value = holder.value;
         if (_isNumber(value)) {
-            return ((NSNumber *)value).floatValue;
+            return ((NSNumber *)value).doubleValue;
```

Comment:
The return type is an `optional<float>` here. I think we should check that `value` is in the range of a float. Do we have an existing method for that?